### ISSUE TITLE
[BUG] tests with hover can be flaky when running in parallel in headed mode

### ISSUE THREAD

**Issue by gurschitz** [microsoft/playwright#27517]
Hey there! We are observing flaky tests related to hovering and I was able to create an example to reproduce it. It's probably not related but it's kind of similar to [this one](https://github.com/microsoft/playwright/issues/27475), as it only seems to be happening in `headed` mode. This time, it's affecting all browsers and it's also a bit less likely to happen.

### System info
- Playwright Version: v1.38.1
- Operating System: Ubuntu 22, MacOS Ventura 13.3.1
- Browser: All
- Other info: I tested this in docker with the latest official docker image `mcr.microsoft.com/playwright:v1.38.1` as well as locally on macOS Ventura 13.3.1.

### Source code

- [x] I provided exact source code that allows reproducing the issue locally.

**Config file**

```ts
// playwright.config.ts
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
  workers: 2,
  projects: [
   {
      name: "chromium",
      use: { ...devices["Desktop Chrome"] },
    },
    {
      name: "firefox",
      use: { ...devices["Desktop Firefox"] },
    },
    {
      name: "webkit",
      use: { ...devices["Desktop Safari"] },
    },
  ]
});
```

**Test file (self-contained)**
This mimics one of our tests. We have a card that reveals a button when the card is hovered, however with a slight delay. 

```ts
test("keeps the hover state", async ({ page }) => {
  await page.setContent(`
  <div data-testid="container" class="container">
    <div>test</div>
    <button type="button">click me</button>
  </div>
 `);

  await page.addStyleTag({
    content: `
    .container {
      position: relative;
      width: 100px;
      height: 100px;
    }

    button {
      display: none;
      position: absolute;
      top: 0;
      right: 0;
    }

    .container:hover button {
      display: block;
    }
  `,
  });
  await page.getByTestId("container").hover();
  
  // even after the timeout, the button should be visible and clickable
  await page.waitForTimeout(1000);

  await page.getByRole("button", { name: "click me" }).click();
});
```

**Steps**
- Start the docker container `docker run --rm --network host -v $(pwd):/work/ -w /work/ -it mcr.microsoft.com/playwright:v1.38.1 /bin/bash`
- Run `xvfb-run npx playwright test --project chromium --headed --repeat-each 20`

**Expected**
- Tests using hover are not flaky when running them with multiple workers in headed mode

**Actual**
Running the above will always have some failing tests for me:

<img width="825" alt="CleanShot 2023-10-09 at 22 24 06@2x" src="https://github.com/microsoft/playwright/assets/4624537/5d4329b4-34f0-4ab0-ab19-6ac80991ede7">

It won't happen in `headless` mode and it also won't happen if I add `--workers 1`. It happens in webkit and firefox as well, however it seems less likely there (I need to repeat them more often to get to failing tests). It also happens outside of docker in macOS. 

Let me know if you need more information, thanks in advance.

**Comment by dgozman** [comment]
I can reproduce on Linux (docker image `mcr.microsoft.com/playwright:next`), but not on MacOS. Interestingly, this only reproduces with `--headed`.

**Comment by prateek-1992** [comment]
I found this while looking for hover action flakiness. Based on the description, I believe this is the cause of the unstable results in our continuous integration pipeline because our tests are designed to automate Chrome extensions, and launching a persistent context requires headless false. If we can get a fix for it soon, it would be of great help

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.

**Comment by Makavelic** [comment]
Can you please look into this? I see similar issues even with headless.

**Comment by vovsemenv** [comment]
still have this issue
